indico/modules/events/editing/controllers/backend.py

```diff
@@ -250,3 +259,23 @@ def _process_PATCH(self, text, internal):
     def _process_DELETE(self):
         delete_revision_comment(self.comment)
         return '', 204
+
+
+class RHExportRevisionFiles(RHContributionEditableRevisionBase):
+    def _check_revision_access(self):
+        return self._user_is_authorized_submitter() or self._user_is_authorized_editor()
+
+    def _process(self):
+        buf = BytesIO()
+        with ZipFile(buf, 'w', allowZip64=True) as zip_handler:
+            for revision_file in self.revision.files:
+                file = revision_file.file
+                filename = secure_filename(file.filename, 'file-{}'.format(file.id))
+                file_type = revision_file.file_type
+                folder_name = secure_filename(file_type.name, 'file-type-{}'.format(file_type.id))
+
+                with file.storage.get_local_path(file.storage_file_id) as filepath:
+                    zip_handler.write(filepath.encode('utf-8'), os.path.join(folder_name, filename))
```

Comment:
is the `.encode('utf-8')` on the local path actually needed? and in case of local storage, `sys.getfilesystemencoding()` would be more appropriate

Reply:
Nope, it is not.